.github/workflows/pull_request.yml

```diff
@@ -36,14 +36,15 @@ jobs:
       - {uses: actions/checkout@v2, with: {fetch-depth: 0}}
       - uses: actions/setup-python@v2
 
-      # Package index is "out of date by design"
-      # https://github.com/actions/virtual-environments/issues/1757
-      - name: apt-get -y update
-        run: sudo apt-get -y update
+      # Use pip to install packages because apt seems to install
+      # the packages for Python whose path is different than setup-python use.
+      # Pylint will report import errors with apt packages
+      # since apt packages can't be found by current Python.
+      - name: update pip
+        run: python -m pip install --upgrade pip
 
       - name: get python libs
-        # FIXME: apt-get succeeds but 'import numpy' still fails!?
-        run: sudo apt-get -y install python3-numpy python3-scipy pylint
+        run: pip install numpy scipy pylint construct graphviz
```

Comment:
Please leave the following line commented out in case someone prefers to use `apt` packages and help Ubuntu fix them. These package names can be difficult to find, also helps other distributions that often use the same names.
```
# sudo apt-get -y install python3-numpy python3-scipy pylint
```

Reply:
Well, should I include the all the packages? like:
```yaml
# sudo apt-get -y install python3-numpy python3-scipy pylint python3-construct python3-graphviz
```

Reply:
Should we just put the dependencies info in [README.md](https://github.com/YongAnLu00/sof-test/blob/tplgtool-ng/README.md#apt-packages)? I think only a few people will check GitHub Actions to find dependencies.

Reply:
If someone tries to report an error reported by CI without looking at the exact commands run by CI then they have a big problem. The commands can often be found in the logs but not always, so in general it's always recommended to look at the actual commands instead of guessing.

On the other hand it shouldn't be required to look at github actions to set up a system (before even running any test).

The ideal is to always have as little code as possible in .github/. Here for instance: `pip3 install -r pip-requirements.txt` or `apt-get install $(cat python-deps.txt)` is ideal. If apt is not possible and pip must be used (not sure about that) then the list of apt packages can be a comment in the `requirements.txt` file.